### PR TITLE
Add skeleton code for ptracepatch pintool

### PR DESCRIPTION
**Pull request by adityavardhanpadala**

### CODE REVIEW

File: Anti2RE/ptracepatch.cpp [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+ * Copyright 2002-2019 Intel Corporation.
+ * 
+ * This software is provided to you as Sample Source Code as defined in the accompanying
+ * End User License Agreement for the Intel(R) Software Development Products ("Agreement")
+ * section 1.L.
+ * 
+ * This software and the related documents are provided as is, with no express or implied
+ * warranties, other than those that are expressly stated in the License.
+ */
+
```
Comment (R3x) [inline Anti2RE/ptracepatch.cpp]: Remove Intel Copyright if you are not using their code. I will add our copyrights later

**Comment by R3x** [comment]
Looks like this doesn't have any functionality yet. Remove all the unnecessary code. And we can merge it.

**Comment by R3x** [comment]
Also missing - Makefile

**Comment by adityavardhanpadala** [comment]
@R3x I modified the skeleton code to trace the function calls and added necessary utility functions. Couldn't add the rest of the code as I was having compile issues will fix and send the changes soon

**Comment by adityavardhanpadala** [comment]
The current output is as follows we just need to find a way to compare the call address and ptrace address. Then we can patch the program in the detach routine.
```
0x7f49516c74ba  call 0x7f49516d14e0
0x7f49516bfed1  call qword ptr [rip+0x1d099]
0x7f49516bff58  call qword ptr [r14]
0x5620d3bb0122  call 0x5620d3bb0040
0x7f493d1641f6  call 0x7f493d1b1a80
0x5620d3bb0127  call 0x5620d3bb0090
0x7f49516bff7b  call rax
0x7f49516bfdc1  call qword ptr [rip+0x1d1a1]
0x7f493d163ace  call 0x7f493d13f330
0x7f493d163b30  call qword ptr [rbx]
0x7f493d1aff04  call 0x7f493d1afbb0
0x7f493d1afc08  call rax
0x7f493d1aff3e  call rax
0x7f493d163b3d  call 0x7f493d200100
Done Execution Ptrace at 7f493d232890
```

**Pull request merged**